pkg/cluster/cluster_config.go

```diff
@@ -127,6 +129,16 @@ func GetPlatform(cli client.Client) (Platform, error) {
 	return isSelfManaged(cli)
 }
 
+func GetPlatform(cli client.Client) (Platform, error) {
+	if cachedPlatform != nil {
+		return *cachedPlatform, nil
+	}
+
+	platform, err := getPlatform(cli)
```

Comment:
i am reading this https://redhat-internal.slack.com/archives/C064Z5S84F3/p1718199614158759
and start to think if we plan to get this platform write into "release" section.
e.g 
name: Red Hat OpenShift AI(Self-managed)
vs
name: Red Hat OpenShift AI(Managed)

Reply:
How you feel about this?  @VaishnaviHire / @etirelli 

and any later call can check dsci.spec.release.name 
and fall back to call getPlatform()